### ISSUE TITLE
[BUG] AzureMonitorExporter's internal MeterProvider is producing confusing self-diagnostic logs

### ISSUE THREAD

**Issue by cijothomas** [Azure/azure-sdk-for-net#42096]
### Library name and version

Azure.Monitor.OpenTelemetry.Exporter 1.2.0

### Describe the bug

https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Statsbeat/AzureMonitorStatsbeat.cs#L71

The Exporter is setting up an internal MeterProvider, leading to potentially misleading internal logs from OpenTelemetry.
For example, if I just enable Warning or above from self-diagnostics, the following internal will be collected. This is very misleading, as this gives the impression that the particular metric will be ignored. But this is not true. The warning is coming from the internal MeterProvider setup by the AzMonExporter (and there will be entry for pretty much every metric.). 
To make things worse, if I enable Verbose level logs, then will see 2 conflicting statements - one saying metric is ignored and the one that says just the opposite!

```
2024-02-21T05:25:37.5119432Z:Measurements from Instrument '{0}', Meter '{1}' will be ignored. Reason: '{2}'. Suggested action: '{3}'{http.server.request.duration}{OpenTelemetry.Instrumentation.AspNetCore}{Instrument belongs to a Meter not subscribed by the provider.}{Use AddMeter to add the Meter to the provider.}
```

This, while causes no issues in the product usability itself, makes it hard to follow the internal logs.

### Expected behavior

No misleading logs caused by/triggered by adding azmon exporter.

### Actual behavior

Adding azmonexporter leds to misleading logs.

### Reproduction Steps

Add AzMonExporter to any .NET app instrumented with OTel metrics. The issue just gave an example.

### Environment

Independent of environment.

**Comment by cijothomas** [comment]
Thanks. I forgot that this was already reported in the otel repo.
Back to azmon - can we not leverage the existing meterprovider from the user itself instead of creating own? (it is also going to create its own thread which is wasted sitting idle for 99.99999% time)

**Comment by vishweshbankwar** [comment]
> Thanks. I forgot that this was already reported in the otel repo. Back to azmon - can we not leverage the existing meterprovider from the user itself instead of creating own? (it is also going to create its own thread which is wasted sitting idle for 99.99999% time)

No - we need to send StatsBeat metrics to a different ikey.

**Comment by cijothomas** [comment]
> > Thanks. I forgot that this was already reported in the otel repo. Back to azmon - can we not leverage the existing meterprovider from the user itself instead of creating own? (it is also going to create its own thread which is wasted sitting idle for 99.99999% time)
> 
> No - we need to send Statsbeat metrics to a different ikey.

Why would that need a new meterprovider? OTel meterprovider supports any number of exporters (though the _current_ internal implementation spins up own thread.), so one can point to IkeyA and another exporter to IkeyB and so on.

**Comment by vishweshbankwar** [comment]
> > > Thanks. I forgot that this was already reported in the otel repo. Back to azmon - can we not leverage the existing meterprovider from the user itself instead of creating own? (it is also going to create its own thread which is wasted sitting idle for 99.99999% time)
> > 
> > 
> > No - we need to send Statsbeat metrics to a different ikey.
> 
> Why would that need a new meterprovider? OTel meterprovider supports any number of exporters (though the _current_ internal implementation spins up own thread.), so one can point to IkeyA and another exporter to IkeyB and so on.

But there isn't way to direct Meter A data goes to ikeyA and MeterB goes to IkeyB? All the data would be sent to both ikeyA and ikeyB right?

**Comment by cijothomas** [comment]
> > > > Thanks. I forgot that this was already reported in the otel repo. Back to azmon - can we not leverage the existing meterprovider from the user itself instead of creating own? (it is also going to create its own thread which is wasted sitting idle for 99.99999% time)
> > > 
> > > 
> > > No - we need to send Statsbeat metrics to a different ikey.
> > 
> > 
> > Why would that need a new meterprovider? OTel meterprovider supports any number of exporters (though the _current_ internal implementation spins up own thread.), so one can point to IkeyA and another exporter to IkeyB and so on.
> 
> But there isn't way to direct Meter A data to go to ikeyA and MeterB to go to IkeyB? All the data would be sent to both ikeyA and ikeyB right?

The AzMonExporter meant for StatsB can ignore every other metric. The AzMonExporter meant for non-statsB can ignore statsb metric.

**Comment by vishweshbankwar** [comment]
> > > > > Thanks. I forgot that this was already reported in the otel repo. Back to azmon - can we not leverage the existing meterprovider from the user itself instead of creating own? (it is also going to create its own thread which is wasted sitting idle for 99.99999% time)
> > > > 
> > > > 
> > > > No - we need to send Statsbeat metrics to a different ikey.
> > > 
> > > 
> > > Why would that need a new meterprovider? OTel meterprovider supports any number of exporters (though the _current_ internal implementation spins up own thread.), so one can point to IkeyA and another exporter to IkeyB and so on.
> > 
> > 
> > But there isn't way to direct Meter A data to go to ikeyA and MeterB to go to IkeyB? All the data would be sent to both ikeyA and ikeyB right?
> 
> The AzMonExporter meant for StatsB can ignore every other metric. The AzMonExporter meant for non-statsB can ignore statsb metric.

This works well when we own both the exporters. We did consider this solution. However, it does not work when other exporters such as otlp is used alongside (We do have this scenario)

**Comment by cijothomas** [comment]
Is that an issue if we sent a metric every 24 hr to the other exporter (s)? if someone is worried about that, they can turn off statsbeat completely (we have off switch right?)

**Comment by cijothomas** [comment]
> This works well when we own both the exporters. 

The exporter is already written with the assumption that we are the only player in the picture (as it overwrites sampling decisions, affecting OTLP exporters or any other exporter/processor). Sending a statsbeat every 24hr seems much less disruptive.

**Comment by vishweshbankwar** [comment]
> Is that an issue if we sent a metric every 24 hr to the other exporter (s)? if someone is worried about that, they can turn off statsbeat completely (we have off switch right?)

Not just statsbeat. We send standard metrics too (different meterprovider for the same reason)

**Comment by TimothyMothra** [comment]
We're going to hold on this topic for now.
There are ongoing discussions about the future direction of Statsbeat.

**Comment by cijothomas** [comment]
> > Is that an issue if we sent a metric every 24 hr to the other exporter (s)? if someone is worried about that, they can turn off statsbeat completely (we have off switch right?)
> 
> Not just statsbeat. We send standard metrics too (different meterprovider for the same reason)

Its another feature which should have on/off capability. And if someone is adding a 3rd party exporter, and don't want it, they can turn off standard metrics.
